sympy/assumptions/sathandlers.py

```diff
@@ -201,28 +188,28 @@ def __call__(self, expr):
 
 ## Abs ##
 
-@class_fact_registry.multiregister(Abs)
-def _(expr):
-    arg = expr.args[0]
-    return [Q.nonnegative(expr),
-            Equivalent(~Q.zero(arg), ~Q.zero(expr)),
-            Q.even(arg) >> Q.even(expr),
-            Q.odd(arg) >> Q.odd(expr),
-            Q.integer(arg) >> Q.integer(expr),
-            ]
+for fact in [
+    lambda expr: Q.nonnegative(expr),
+    lambda expr: Equivalent(~Q.zero(expr.args[0]), ~Q.zero(expr)),
+    lambda expr: Q.even(expr.args[0]) >> Q.even(expr),
+    lambda expr: Q.odd(expr.args[0]) >> Q.odd(expr),
+    lambda expr: Q.integer(expr.args[0]) >> Q.integer(expr),
+]:
+    class_fact_registry.register(Abs)(fact)
 
 
 ### Add ##
 
-@class_fact_registry.multiregister(Add)
-def _(expr):
-    return [allargs(x, Q.positive(x), expr) >> Q.positive(expr),
-            allargs(x, Q.negative(x), expr) >> Q.negative(expr),
-            allargs(x, Q.real(x), expr) >> Q.real(expr),
-            allargs(x, Q.rational(x), expr) >> Q.rational(expr),
-            allargs(x, Q.integer(x), expr) >> Q.integer(expr),
-            exactlyonearg(x, ~Q.integer(x), expr) >> ~Q.integer(expr),
-            ]
+for fact in [
+    lambda expr: allargs(x, Q.positive(x), expr) >> Q.positive(expr),
+    lambda expr: allargs(x, Q.negative(x), expr) >> Q.negative(expr),
+    lambda expr: allargs(x, Q.real(x), expr) >> Q.real(expr),
+    lambda expr: allargs(x, Q.rational(x), expr) >> Q.rational(expr),
+    lambda expr: allargs(x, Q.integer(x), expr) >> Q.integer(expr),
+    lambda expr: exactlyonearg(x, ~Q.integer(x), expr) >> ~Q.integer(expr)
+]:
+    class_fact_registry.register(Add)(fact)
```

Comment:
It's not really clear to me how this is an improvement.

Reply:
`multiregister` method is removed since this change expects one-to-one relations between the handlers and the facts, in order to pick a handler which returns the desired fact.